hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -4550,6 +4619,10 @@ private void doMiniBatchMutate(BatchOperation<?> batchOp) throws IOException {
     /** Keep track of the locks we hold so we can release them in finally clause */
     List<RowLock> acquiredRowLocks = Lists.newArrayListWithCapacity(batchOp.size());
     try {
+      // Check for thread interrupt status in case we have been signaled from
+      // #interruptRegionOperation.
+      checkInterrupt();
```

Comment:
Pity we need to repeat this call many times in the same method...

Reply:
Yes, but on the other hand we are basically checking interrupt status manually, as one has to do in Java, so can check and exit early only at only what we think are safe points.